editor/settings_config_dialog.cpp

```diff
@@ -508,11 +508,11 @@ EditorSettingsDialog::EditorSettingsDialog() {
 
 	timer = memnew(Timer);
 	timer->set_wait_time(1.5);
-	timer->connect("timeout", this, "_settings_save");
+	timer->connect("timeout", EditorSettings::get_singleton(), "_settings_save");
 	timer->set_one_shot(true);
 	add_child(timer);
-	EditorSettings::get_singleton()->connect("settings_changed", this, "_settings_changed");
-	get_ok()->set_text(TTR("Close"));
+	//EditorSettings::get_singleton()->connect("settings_changed", this, "_settings_changed");
```

Comment:
Will this cause the editor settings change not to be applied automatically, only saved?

Reply:
Yes that's what it seems to be doing

Reply:
It will be applied when you press the button ok at the bottom of the editor setting .

Reply:
It would be better if the automatic application could be preserved, but this fix is IMO better than the current behaviour, since picking a color for themes is very annoying with the current behaviour.

The fix here also affects properties that do not use the color picker like 'Contrast', though I do not at the moment see an easy way to only affect the color picker.

Preserving the automatic application might be a bit to somewhat more involved; when I looked into it, this comment is as far as I got: https://github.com/godotengine/godot/issues/23626#issuecomment-473516855 .

Reply:
Thinking about it, your solution might be the best option overall, because otherwise there might be a variety of cyclic effects, such as is seen with the color picker. For instance: Some property is changed, which shortly afterwards affects the GUI of the property again, potentially while editing that property. One could in theory make the system robust against that, but as far as I can figure out, that may take a fair bit of work at least, and it would not give a large return on investment, since this is just editor settings.

Reply:
Hi,
How can i delete some of the commits from this pull request and only leave commit a675968 ? I am new to git and I haven't mastered its functionalities yet.

Reply:
Have you tried finding a tutorial online specifically for that? There should be some good ones online, and StackOverflow should also have some good answers, since people have encountered this problem before a lot.